From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas <lukasalt98@gmail.com>
Date: Sun, 27 Dec 2020 01:09:16 +0100
Subject: [PATCH] added option to disable elytras


diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
index fc711a596bc424b31739a29a222325438b644882..3351a86c103806b0b7f13d03f0328ac2b4050835 100644
--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
@@ -723,4 +723,9 @@ public class PaperWorldConfig {
     private void mapInItemFrameSettings() {
         mapInItemFrameUpdateInterval = Math.max(1, getInt("map-in-item-frame-update-interval", 10));
     }
+    
+    public boolean disableElytraInEndCities = false;
+    private void endCitySettings() {
+        disableElytraInEndCities = getBoolean("disable-elytra-in-end-cities", this.disableElytraInEndCities);
+    }
 }
diff --git a/src/main/java/net/minecraft/server/WorldGenEndCityPieces.java b/src/main/java/net/minecraft/server/WorldGenEndCityPieces.java
index 45aecd255a1ff28f2b22fd3dfa6ff4bcb34db7af..b6b2a5e78ce47463bedfda1eab78a63ff8221717 100644
--- a/src/main/java/net/minecraft/server/WorldGenEndCityPieces.java
+++ b/src/main/java/net/minecraft/server/WorldGenEndCityPieces.java
@@ -279,7 +279,7 @@ public class WorldGenEndCityPieces {
                 entityshulker.setPosition((double) blockposition.getX() + 0.5D, (double) blockposition.getY() + 0.5D, (double) blockposition.getZ() + 0.5D);
                 entityshulker.h(blockposition);
                 worldaccess.addEntity(entityshulker);
-            } else if (s.startsWith("Elytra")) {
+            } else if (s.startsWith("Elytra") && !worldaccess.getMinecraftWorld().paperConfig.disableElytraInEndCities) { // Paper - added option to elytra
                 EntityItemFrame entityitemframe = new EntityItemFrame(worldaccess.getMinecraftWorld(), blockposition, this.e.a(EnumDirection.SOUTH));
 
                 entityitemframe.setItem(new ItemStack(Items.ELYTRA), false);
