From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gmail.com>
Date: Sat, 2 Jan 2021 17:13:43 +0100
Subject: [PATCH] Optimized chunk lookups for entities


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index a6520f80b873c348af23fb50435f6a16736abb6e..14a427ac33d46de6a8ce6c6cfb14634b2c8cb323 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -810,7 +810,7 @@ public class WorldServer extends World implements GeneratorAccessSeed {
     }
 
     public void entityJoinedWorld(Entity entity) {
-        if (!(entity instanceof EntityHuman) && !this.getChunkProvider().a(entity)) {
+        Chunk chunk; PlayerChunk playerChunk; if (!(entity instanceof EntityHuman) || (chunk = entity.getCurrentChunk()) == null || (playerChunk = chunk.playerChunk) == null || !playerChunk.isEntityTickingReady()) { // Paper - inlined ChunkProviderServer.a(ChunkCoordIntPair)
             this.chunkCheck(entity);
         } else {
             ++TimingHistory.entityTicks; // Paper - timings
