From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gmail.com>
Date: Sun, 3 Jan 2021 21:04:29 +0100
Subject: [PATCH] GHD - optimized light engine queueing


diff --git a/src/main/java/net/minecraft/server/ChunkProviderServer.java b/src/main/java/net/minecraft/server/ChunkProviderServer.java
index 4b8ebc47fa358b3e2ae879358f6b0cc3a8de5dbc..7cab4c5f7e5ae0dbc360f2ba856621b3ed0ead09 100644
--- a/src/main/java/net/minecraft/server/ChunkProviderServer.java
+++ b/src/main/java/net/minecraft/server/ChunkProviderServer.java
@@ -843,6 +843,7 @@ public class ChunkProviderServer extends IChunkProvider {
                     }
                 }
             });
+            getLightEngine().queueUpdate(); // Paper - moved from WorldServer.a(Chunk,int)
             this.world.getMethodProfiler().enter("customSpawners");
             if (flag1) {
                 try (co.aikar.timings.Timing ignored = this.world.timings.miscMobSpawning.startTiming()) { // Paper - timings
diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 2b94c195db43d8e7fd58301a33377e87daa16e98..5ea69cb8abe7da954b84db08ec0fcf1cec41b150 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -715,7 +715,7 @@ public class WorldServer extends World implements GeneratorAccessSeed {
             }
             gameprofilerfiller.exit();
             timings.chunkTicksBlocks.stopTiming(); // Paper
-            getChunkProvider().getLightEngine().queueUpdate(); // Paper
+            // getChunkProvider().getLightEngine().queueUpdate(); // Paper - moved to ChunkProviderServer
             // Paper end
         }
     }
