From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Mon, 25 Jan 2021 14:37:57 +0100
Subject: [PATCH] GHD - added option to disable pathfinding updates on block
 changes


diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
index 0b9dac134d5f7cbfb152ebb3ea2dcb55a0ad7499..1d93e863d8247b2cf9050aa107da2f34ebc37f51 100644
--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
@@ -818,4 +818,9 @@ public class PaperWorldConfig {
     private void setCheckDespawnTickInterval() {
         checkDespawnTickInterval = getInt("check-despawn-tick-interval", this.checkDespawnTickInterval);
     }
+
+    public boolean updatePathfindingOnBlockUpdate = true;
+    private void setUpdatePathfindingOnBlockUpdate() {
+        updatePathfindingOnBlockUpdate = getBoolean("update-pathfinding-on-block-update", this.updatePathfindingOnBlockUpdate);
+    }
 }
diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index ee1b3a1e12875adfc45277afcfcaef441845e865..85182c36f42ddf9b633f6deeec4823e37b4ad836 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -1571,6 +1571,7 @@ public class WorldServer extends World implements GeneratorAccessSeed {
     @Override
     public void notify(BlockPosition blockposition, IBlockData iblockdata, IBlockData iblockdata1, int i) {
         this.getChunkProvider().flagDirty(blockposition);
+        if(!this.paperConfig.updatePathfindingOnBlockUpdate) return; // Paper - option to disable pathfinding updates
         VoxelShape voxelshape = iblockdata.getCollisionShape(this, blockposition);
         VoxelShape voxelshape1 = iblockdata1.getCollisionShape(this, blockposition);
 
